docs/server-dependencies.md

```diff
@@ -0,0 +1,102 @@
+---
+id: server-dependencies
+title: Temporal server dependencies
+sidebar_label: Dependencies
+---
+
+## Overview
+
+This page details some of the requirements and dependencies needed to build and run an instance of Temporal.
+
+All versions of Temporal require the following:
+
+1. [Go 1.15+](https://golang.org/dl/)
+2. [Go package dependencies](https://github.com/temporalio/temporal/blob/master/go.mod)
```

Comment:
the url is pointing to the master branch
for each release, there is different set of dependencies
v1.2.2: https://github.com/temporalio/temporal/blob/v1.2.2/go.mod
v1.3.2: https://github.com/temporalio/temporal/blob/v1.3.2/go.mod